engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -278,6 +279,55 @@ class WorkflowManagerActor(params: WorkflowManagerActorParams)
       logger.debug(s"$tag transitioning from $fromState to $toState")
   }
 
+  private val blacklistCacheConfig: Option[Config] = config.as[Option[Config]]("call-caching.blacklist-cache")
+
+  private val blacklistGroupKey: Option[String] = blacklistCacheConfig.flatMap(_.as[Option[String]]("grouping-workflow-option"))
+
+  // If configuration allows, build a cache of blacklist groupings to BlacklistCaches.
+  private val blacklistGroupings: Option[LoadingCache[String, BlacklistCache]] = {
+    def buildBlacklistGroupings(cacheConfig: CacheConfig): LoadingCache[String, BlacklistCache] = {
+      val emptyBlacklistCacheLoader = new CacheLoader[String, BlacklistCache]() {
+        override def load(key: String): BlacklistCache = BlacklistCache(cacheConfig)
+      }
+
+      CacheBuilder.
+        newBuilder().
+        concurrencyLevel(cacheConfig.concurrency).
+        maximumSize(cacheConfig.size).
```

Comment:
TOL: Since we're using `cacheConfig.size` for both. is it reasonable to assume that "maximum number of buckets in the blacklist" is approximately the same size as "number of submissions currently running workflows". 

Probably yes, since they're ultimately drawing from the same set of values (ie google projects in Terra)?

Reply:
I reused these values purely out of expedience. We should probably think a bit more about appropriate relative sizes, preferably aided by metrics. 🙂